### DISCUSSION TITLE
suggest convert( ... copy=True )

### DISCUSSION THREAD

**Discussion by denis-bz** [coloria-dev/coloria#200] [General]
Nico, just a suggestion: add `copy=True`  or else a warning in the doc,
what do you think ?
```
def rgb_to_cspace( rgb: "3xn", cspace, copy=True, scale=True ) -> "conv.data 3xn":
    if copy:
        rgb = rgb.copy()
    else:
        rgb.setflags( write=False )  # ❓
    conv = cs.ColorCoordinates( rgb, SRGB1() )
    conv.convert( cspace )
    return conv.data
```

**Comment by nschloe** [comment]
I don't quite understand. There's no function `rgb_to_cspace` in colorio.

**Reply by denis-bz** [reply]
sorry, wasn't clear. Is it true that inputs to `ColorCoordinates` and `convert` are not read-only, may be modified ?
Ifso I'd suggest adding a line to that effect to the doc.
Testcase:

```
import numpy as np
from colorio import cs
from colorio.cs import *

rgb = np.eye(3)
rgb.setflags( write=False )  # read-only ?

conv = cs.ColorCoordinates( rgb, SRGB1() )
conv.convert( OKLAB() )  # <-- ValueError: assignment destination is read-only

**Comment by nschloe** [comment]
> Is it true that inputs to ColorCoordinates and convert are not read-only, may be modified ?

Do you mean if they _are_ modified inside `convert()`?
